src/default_metrics.go

```diff
@@ -0,0 +1,127 @@
+	"db.innodb.rowLockWaitsPerSecond":             {"Innodb_row_lock_waits", metric.RATE},
+	"db.openFiles":                                {"Open_files", metric.GAUGE},
+	"db.openTables":                               {"Open_tables", metric.GAUGE},
+	"db.openedTablesPerSecond":                    {"Opened_tables", metric.RATE},
+	"db.tablesLocksWaitedPerSecond":               {"Table_locks_waited", metric.RATE},
+	"software.edition":                            {"version_comment", metric.ATTRIBUTE},
+	"software.version":                            {"version", metric.ATTRIBUTE},
+	"cluster.nodeType":                            {"node_type", metric.ATTRIBUTE},
+}
+
+var defaultMetrics560 = map[string][]interface{}{
+	"db.qCacheFreeMemoryBytes":    {"Qcache_free_memory", metric.GAUGE},
+	"db.qCacheNotCachedPerSecond": {"Qcache_not_cached", metric.RATE},
+	"db.qCacheUtilization":        {qCacheUtilization, metric.GAUGE},
+	"db.qCacheHitRatio":           {qCacheHitRatio, metric.GAUGE},
+	// If a cluster instance is not a slave, then the metric cluster.slaveRunning will be removed.
+	"cluster.slaveRunning": {slaveRunningAsNumber, metric.GAUGE},
+}
+
+var defaultMetrics800 = map[string][]interface{}{
+	// If a cluster instance is not a slave, then the metric cluster.slaveRunning will be removed.
+	"cluster.slaveRunning": {slaveRunningAsNumber, metric.GAUGE},
+}
+
+func slaveRunningAsNumber(metrics map[string]interface{}, version *semver.Version) (int, bool) {
+	var prefix string
+	if version.GE(semver.Version{Major: 8, Minor: 0, Patch: 0}) {
+		prefix = "Replica"
+	} else {
+		prefix = "Slave"
+	}
+	slaveIORunning, okIO := metrics[prefix+"_IO_Running"].(string)
+	slaveSQLRunning, okSQL := metrics[prefix+"_SQL_Running"].(string)
+	if !okIO || !okSQL {
+		return 0, false
+	}
+	if slaveIORunning == "Yes" && slaveSQLRunning == "Yes" {
+		return 1, true
+	}
+	return 0, true
+}
+
+func qCacheUtilization(metrics map[string]interface{}) (float64, bool) {
+	// TODO compute the value within the interval
+	qCacheFreeBlocks, ok1 := metrics["Qcache_free_blocks"].(int)
+	qCacheTotalBlocks, ok2 := metrics["Qcache_total_blocks"].(int)
+
+	if qCacheTotalBlocks == 0 {
+		return 0, true
+	} else if ok1 && ok2 {
+		return 1 - (float64(qCacheFreeBlocks) / float64(qCacheTotalBlocks)), true
+	}
+	return 0, false
+}
+
+func qCacheHitRatio(metrics map[string]interface{}) (float64, bool) {
+	qCacheHits, ok1 := metrics["Qcache_hits"].(int)
+	queries, ok2 := metrics["Queries"].(int)
+
+	if queries == 0 {
+		return 0, true
+	} else if ok1 && ok2 {
+		return float64(qCacheHits) / float64(queries), true
+	}
+	return 0, false
+}
+
+var defaultMetricsVersionDefinitions = []VersionDefinition{
+	{
+		minVersion:        semver.MustParse("8.0.0"),
+		metricsDefinition: mergeMaps(defaultMetricsBase, defaultMetrics800),
+	},
+	{
+		minVersion:        semver.MustParse("5.6.0"),
+		metricsDefinition: mergeMaps(defaultMetricsBase, defaultMetrics560),
```

Comment:
Can we use struct embedding rather than merging structures?

Reply:
proceeding with using mergeMaps as we cannot use embedding with maps. 

Resolved it in this [commit](https://github.com/newrelic/nri-mysql/pull/170/commits/fe29f92e5ad2d923fd031d278d2ecb482b0e69be)